src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Management.Automation.Internal;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NJsonSchema;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// This class implements Test-Json command.
+    /// </summary>
+    [Cmdlet(VerbsDiagnostic.Test, "Json", HelpUri = "")]
+    public class TestJsonCommand : PSCmdlet
+    {
+        /// <summary>
+        /// An Json to be validated.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true)]
+        public String Json { get; set; }
+
+        /// <summary>
+        /// A schema to validate the Json against.
+        /// It is optional parameter.
+        /// If the parameter is absent the cmdlet only try to parse the Json.
+        /// If the parameter present the cmdlet try to parse the Json and
+        /// then check the Json against the schema. Before the check
+        /// the cmdlet parse the schema doing implicitly check the schema too.
+        /// </summary>
+        [Parameter(Position = 1)]
+        [ValidateNotNullOrEmpty()]
```

Comment:
Why disallow $null?  This makes it hard to standardize calls to cmdlet.

Reply:
I was thinking that we could enhance this cmdlet in the future to explicitly check the schema but haven't analyzed yet. So I blocked this to avoid breaking changes in the future. I would like to get more feedback from users before doing enhancements to test JSON schema - yet this cmdlet is designed to test first of all the JSON not JSON schema.

Another consideration is that these are too different scenarios (checking without schema and checking with schema) it is unlikely that they will conflict.

/cc @dantraMSFT @SteveL-MSFT Could you please comment too?